### PR TITLE
Feature/mail confirmation

### PR DESCRIPTION
**Pull request by airat-git**

### CODE REVIEW

File: auth/src/main/java/com/emporium/auth/service/UserServiceImpl.java [hunk 1]
```diff
@@ -53,6 +59,39 @@ public String create(RegistrationDTO dto) {
...
+        if (optionalUser.isEmpty()){
+            log.error("Failed to find user by id. id: {}",id);
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "User not found.");
+        }
+        log.debug("findById() - end. user: {}", optionalUser.get());
+        return optionalUser.get();
+    }
+
+    @Override
+    public void mailConfirm(ObjectId id) {
```
Comment (IlyaOkonechnikov) [inline auth/src/main/java/com/emporium/auth/service/UserServiceImpl.java:87]: confirmMail

File: auth/src/main/java/com/emporium/auth/service/UserServiceImpl.java [hunk 2]
```diff
@@ -53,6 +59,39 @@ public String create(RegistrationDTO dto) {
         }
     }
 
+    @Override
+    public User findByEmail(String email) {
```
Comment (IlyaOkonechnikov) [inline auth/src/main/java/com/emporium/auth/service/UserServiceImpl.java:63]: Удоли

File: auth/src/main/java/com/emporium/auth/controller/AuthController.java [hunk 3]
```diff
@@ -26,4 +27,11 @@ public String register(@RequestBody @Valid RegistrationDTO dto) {
     public String login(@RequestBody @Valid RegistrationDTO dto) {
         return null;
     }
+
+    @GetMapping("/mail-confirmation/{id}")
```
Comment (IlyaOkonechnikov) [inline auth/src/main/java/com/emporium/auth/controller/AuthController.java:31]: Patch

**Review by IlyaOkonechnikov — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/emporium/auth/config/MailConfirmationConfig.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    private final String from;
+    private final String socketFactoryPort;
+    private final String socketFactoryClass;
+    private final Properties property ;
+
+    public MailConfirmationConfig(@Value("${mail.host}") String mailHost,
+                                  @Value("${mail.port}") String mailPort,
+                                  @Value("${mail.username}")String mailUsername,
+                                  @Value("${mail.password}")String mailPassword,
+                                  @Value("${mail.from}")String from,
```
[the remark this answers is not in the document]
Comment (airat-git) [reply auth/src/main/java/com/emporium/auth/config/MailConfirmationConfig.java:27]: Habib Case?)

[the remark this answers is not in the document]
Comment (IlyaOkonechnikov) [reply auth/src/main/java/com/emporium/auth/config/MailConfirmationConfig.java]: Habib Win)

**Review by IlyaOkonechnikov — APPROVED** [review]

**Pull request merged**